st2common/bin/st2ctl

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env bash
+
+if [ $(id -u) -ne 0 ]; then
+    echo "Please run with root privileges"
+    exit 1
+fi
+
+LOGFILE="/dev/null"
+COMPONENTS="st2actionrunner st2api st2auth st2garbagecollector st2notifier st2resultstracker st2rulesengine st2sensorcontainer st2web mistral"
+STANCONF="/etc/st2/st2.conf"
+WEBUI_PORT=${WEBUI_PORT:-8080}
+
+# Ensure global environment is sourced if exists
+# Does not happen consistently with all OSes we support.
+[ -r /etc/environment ] && . /etc/environment
+
+
+function print_usage() {
+    echo "Usage: st2ctl {start, stop, restart, status}"
+    echo
+    echo "Usage: st2ctl {restart-component}"
+    echo "positional arguments:"
+    echo "  component           Name of the st2 service to restart."
+    echo "                      ${COMPONENTS}"
+    echo
+    echo "Usage: st2ctl {reopen-log-files}"
+    echo "positional arguments:"
+    echo "  component           Name of the st2 service to reopen the log files for."
+    echo "                      ${COMPONENTS}"
+    echo
+    echo "Usage: st2ctl {reload, clean}"
+    echo "optional arguments:"
+    echo "  --register-all      Register all."
+    echo "  --register-sensors  Register all sensors."
+    echo "  --register-rules    Register all rules."
+    echo "  --register-actions  Register all actions."
+    echo "  --register-aliases  Register all aliases."
+    echo "  --register-policies Register all policies."
+    echo "  --verbose           Output additional debug and informational messages."
+}
+
+function validate_in_components() {
+  COM=${1}
+
+  if [ -z ${COM} ]; then
+    echo -e "\e[31mError: Component is required! \e[0m\n"
+    print_usage
+    exit 1
+  fi
+
+  if [[ " ${COMPONENTS} " != *" ${COM} "* ]]; then
+    echo -e "\e[31mError: Invalid component provided: ${COM} \e[0m\n"
+    print_usage
+    exit 1
+  fi
+}
+
+function st2start() {
+  for COM in ${COMPONENTS}; do
+    service ${COM} start
+  done
+}
+
+function st2stop() {
+  for COM in ${COMPONENTS}; do
+    service ${COM} stop
+  done
+}
+
+# Next candidate for removal
+# after we implement native `service st2component reopen` (nginx example) for each component
+function reopen_component_log_files() {
```

Comment:
Used in [`logrotate.conf`](https://github.com/StackStorm/st2/blob/2f6465f366bc22c8adff79c031acd0a3b3837728/conf/logrotate.conf#L11)

See bash comments.